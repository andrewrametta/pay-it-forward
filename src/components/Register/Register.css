.register-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.register-item {
  width: 80%;
  margin: 0 auto;
}
