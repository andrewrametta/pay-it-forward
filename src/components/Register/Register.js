import React from "react";
import "./Register.css";

function Register() {
  return (
    <div className="register-wrapper">
      <section className="register-section">
        <header className="register-header">
          <h1>Register</h1>
        </header>
        <article>
          <div>
            <label className="label" htmlFor="usertype">
              User Type
            </label>
            <select id="usertype" name="usertype">
              <option value="donor">Donor</option>
              <option value="organization">Organization</option>
            </select>
          </div>
          <div>
            <form className="register form">
              <div>
                <label className="label" htmlFor="username">
                  Username
                </label>
                <input
                  type="text"
                  name="username"
                  id="username"
                  placeholder="username"
                />
              </div>
              <div>
                <label className="label" htmlFor="email">
                  Email
                </label>
                <input
                  type="email"
                  name="email"
                  id="email"
                  placeholder="email"
                />
              </div>
              <div>
                <label className="label" htmlFor="password">
                  Password
                </label>
                <input
                  type="text"
                  name="password"
                  id="password"
                  placeholder="password"
                />
              </div>
              <div className="register-button">
                <button>Register</button>
              </div>
            </form>
          </div>
        </article>
      </section>
    </div>
  );
}

export default Register;
