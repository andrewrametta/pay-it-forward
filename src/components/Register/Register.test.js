import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Register from "./Register";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe(`Register component`, () => {
  it("renders the complete page", () => {
    const wrapper = shallow(<Register />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
