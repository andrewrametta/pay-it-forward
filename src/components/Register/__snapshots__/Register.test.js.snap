// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders the complete page 1`] = `
<div
  className="register-wrapper"
>
  <section
    className="register-section"
  >
    <header
      className="register-header"
    >
      <h1>
        Register
      </h1>
    </header>
    <article>
      <div>
        <form
          className="register form"
          onSubmit={[Function]}
        >
          <div>
            <label
              className="label"
              htmlFor="usertype"
            >
              User Type
            </label>
            <select
              id="usertype"
              name="usertype"
            >
              <option
                value="user"
              >
                User
              </option>
              <option
                value="org"
              >
                Organization
              </option>
            </select>
          </div>
          <div>
            <label
              className="label"
              htmlFor="username"
            >
              Username or Organization Name
            </label>
            <input
              id="username"
              name="username"
              placeholder="username"
              type="text"
            />
          </div>
          <div>
            <label
              className="label"
              htmlFor="email"
            >
              Email
            </label>
            <input
              id="email"
              name="email"
              placeholder="email"
              type="email"
            />
          </div>
          <div>
            <label
              className="label"
              htmlFor="password"
            >
              Password
            </label>
            <input
              id="password"
              name="password"
              placeholder="password"
              type="password"
            />
          </div>
          <div>
            <label
              className="label"
              htmlFor="address"
            >
              Address
            </label>
            <input
              id="address"
              name="address"
              placeholder="address"
              type="text"
            />
          </div>
          <div>
            <label
              className="label"
              htmlFor="city"
            >
              City
            </label>
            <input
              id="city"
              name="city"
              placeholder="city"
              type="text"
            />
          </div>
          <div>
            <label
              className="label"
              htmlFor="state"
            >
              state
            </label>
            <input
              id="state"
              name="state"
              placeholder="ST"
              type="text"
            />
          </div>
          <div>
            <label
              className="label"
              htmlFor="zipcode"
            >
              zipcode
            </label>
            <input
              id="zipcode"
              name="zipcode"
              pattern="[0-9]{5}"
              placeholder="zipcode"
              title="Five digit zip code"
              type="text"
            />
          </div>
          <div
            className="register-button"
          >
            <button
              type="submit"
            >
              Register
            </button>
          </div>
        </form>
      </div>
    </article>
  </section>
</div>
`;
